tests/test_api.py

```diff
@@ -1,119 +1,62 @@
-
-def test_dbf_file_exist():
-    filepath=__location__+"/Testdata/Abgrabungen_Kreis_Kleve_Shape.dbf"
-    assert os.path.isfile(filepath) == True
-
-###############################
-# Check bbox extraction
-###############################
 def test_geojson_extract_bbox():
-    filepath=__location__+"/Testdata/muenster_ring_zeit.geojson"
-    geojson = pygeoj.load(filepath)
-    bbox = (geojson).bbox
-    assert bbox == [7.6016807556152335, 51.94881477206191, 7.647256851196289, 51.974624029877454]
+    filepath=__location__+"/Testdata/folder/muenster_ring_zeit.geojson"
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[7.6016807556152335, 51.94881477206191, 7.647256851196289, 51.974624029877454],['2018-11-14']]
 
 def test_invalid_coordinate_geojson_extract_bbox(capsys):
     filepath=__location__+"/Testdata/invalid_coordinate.geojson"
-    sys.stderr.write("error message")
-    print("Invalid geojson file")
-    out, error = capsys.readouterr()
-    assert out == "Invalid geojson file\n"
-    assert error == "error message"
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[None],[None]]
+
 
 def test_one_point_geojson_extract_bbox():
     filepath=__location__+"/Testdata/onePoint.geojson"
-    geojson = pygeoj.load(filepath)
-    bbox = (geojson).bbox
-    assert bbox == [6.220493316650391, 50.52150360276628, 6.220493316650391, 50.52150360276628]
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[6.220493316650391, 50.52150360276628, 6.220493316650391, 50.52150360276628],[None]]
 
 def test_empty_file_geojson_extract_bbox(capsys):
     filepath=__location__+"/Testdata/empty.geojson"
-    captured = capsys.readouterr()
-    assert captured.out == "Empty file"
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[None],[None]]
+
 
-def test_nc_extract_bbox():
+def test_netcdf_extract_bbox():
     filepath=__location__+"/Testdata/ECMWF_ERA-40_subset1.nc"
-    assert netcdf_bbox(filepath) == [0.0, -90.0, 357.5, 90.0]
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[-90.0, 0.0, 90.0, 357.5],['2002-07-31']]
+
+
+def test_kml_extract_bbox():
+    filepath=__location__+"/Testdata/aasee.kml"
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[7.594213, 51.942466, 7.618246, 51.957278],[None]]
+
+def test_geotiff_extract_bbox():
+    filepath=__location__+"/Testdata/wf_100m_klas.tif"
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[5.9153007564753155, 50.31025197410836, 9.468398712484145, 52.5307755328733],[None]]
+
+def test_gpkg_extract_bbox():
+    filepath=__location__+"/Testdata/census2016_cca_qld_short.gpkg"
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[-43.7405, 96.8169, -9.14218, 167.998],[None]]
+
+def test_csv_extract_bbox():
+    filepath=__location__+"/Testdata/folder/cities_NL.csv"
+    assert geoextent.getMetadata(filepath, 'bbox' , True) == [[6.574722, 51.434444, 4.3175, 53.217222],['2018-09-30']]]
```

Comment:
Please add different time stamps to https://github.com/o2r-project/geoextent/blob/39822dedba48684f4aa3a450811ed3cba158b4ec/tests/Testdata/folder/cities_NL.csv so you get an interval here.

Please also create a test file that has invalid time stamps, and time stamps that include the time. I think CSV is a reasonable format to test these variations as they are easy to create. Other formats then only need to test that they find the timestamp column.